cedar-policy-validator/src/typecheck.rs

```diff
@@ -1388,12 +1422,22 @@ impl<'a> Typechecker<'a> {
             }
 
             BinaryOp::Add | BinaryOp::Sub => {
+                let help_builder = |actual: &Type| match (op, actual) {
+                    (
+                        BinaryOp::Add,
+                        Type::Primitive {
+                            primitive_type: Primitive::String,
+                        },
+                    ) => Some(UnexpectedTypeHelp::ConcatenationNotSupported),
```

Comment:
I would suggest also throwing `ConcatenationNotSupported` in the case of `+` on sets, maybe also records

Reply:
Also makes sense to say some on `-` for sets